plugins/extractors/maxcompute/maxcompute.go

```diff
@@ -0,0 +1,251 @@
+package maxcompute
+
+import (
+	"context"
+	_ "embed" // used to print the embedded assets
+	"sync"
+	"time"
+
+	client2 "github.com/alibabacloud-go/darabonba-openapi/v2/client"
+	maxcomputeclient "github.com/alibabacloud-go/maxcompute-20220104/client"
+	"github.com/goto/meteor/models"
+	v1beta2 "github.com/goto/meteor/models/gotocompany/assets/v1beta2"
+	"github.com/goto/meteor/plugins"
+	"github.com/goto/meteor/registry"
+	"github.com/goto/meteor/utils"
+	"github.com/goto/salt/log"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type Config struct {
+	ProjectName     string `mapstructure:"project_name"`
+	EndpointProject string `mapstructure:"endpoint_project"`
+	AccessKey       struct {
+		ID     string `mapstructure:"id"`
+		Secret string `mapstructure:"secret"`
+	} `mapstructure:"access_key"`
+	SchemaName     string `mapstructure:"schema_name"`
+	MaxPreviewRows int    `mapstructure:"max_preview_rows"`
+	Exclude        struct {
+		Schemas []string `mapstructure:"schemas"`
+		Tables  []string `mapstructure:"tables"`
+	} `mapstructure:"exclude"`
+	MaxPageSize          int32    `mapstructure:"max_page_size"`
+	Concurrency          int      `mapstructure:"concurrency"`
+	MixValues            bool     `mapstructure:"mix_values"`
+	IncludeColumnProfile bool     `mapstructure:"include_column_profile"`
+	BuildViewLineage     bool     `mapstructure:"build_view_lineage"`
+	IsCollectTableUsage  bool     `mapstructure:"collect_table_usage"`
+	UsagePeriodInDay     int      `mapstructure:"usage_period_in_day"`
+	UsageProjectNames    []string `mapstructure:"usage_project_names"`
+}
+
+type Extractor struct {
+	plugins.BaseExtractor
+	logger log.Logger
+	config Config
+}
+
+//go:embed README.md
+var summary string
+
+var sampleConfig = `
+project_name: goto_test
+endpoint_project: http://goto_test-maxcompute.com
+access_key:
+    id: access_key_id
+    secret: access_key_secret
+schema_name: DEFAULT
+max_preview_rows: 3
+exclude:
+	schemas:
+	- schema_a
+	- schema_b
+	tables:
+	- schema_c.table_a
+max_page_size: 100
+concurrency: 10
+mix_values: false
+include_column_profile: true
+build_view_lineage: true
+collect_table_usage: false
+usage_period_in_day: 7
+usage_project_names:
+	- maxcompute-project-name
+	- other-maxcompute-project-name`
```

Comment:
please fix the format, also have we implement all of this config?

Reply:
addressed